src/CMakeLists.txt

```diff
@@ -130,10 +103,6 @@ set (NN_SOURCES
     utils/sleep.c
     utils/thread.h
     utils/thread.c
-    utils/thread_posix.h
```

Comment:
The reason I chose this method is to prune other-platform source code from the backend-generated project file -- in my case this is a Visual Studio Solution. By omitting, e.g., `worker_posix.inc`, this means when I search for symbols, it doesn't return duplicates in non-relevant platforms -- only `worker_win.inc` was included and relevant. It makes the IDE experience better.

That said ... there's merit in understanding other platforms, even if it's not relevant to your own current platform. In which case, I would prefer a rearchitecture of the source code to be along the lines of this: https://github.com/nanomsg/nanomsg/blob/master/src/utils/clock.c#L71-L121

Either way ... much of this entire CMake rearchitecture is a stepping stone to get from where we are to where we want to be, and in the meantime, it's likely to carry along some cruft that can then be hit later.